src/inject/dynamic-theme/style-manager.ts

```diff
@@ -53,7 +53,11 @@ export function shouldManageStyle(element: Node) {
             )
         ) &&
         !element.classList.contains('darkreader') &&
-        element.media !== 'print' &&
+        (
+            element.media !== 'print' ||
+            // Some stylesheets uses 'lazy loading' and has on initial a `print` media.
+            element.getAttribute('onload').toLowerCase().includes('this.media=')
```

Comment:
Ehmm, looks like a hack. Is it a common problem for these websites? Why do they do this? Like the styles are not shown until they finish loading? I thought the browser optimizes it. Maybe it fixes the problem for some IE.

Let's find out if there's a less hacky way. E.g. watching for media attribute change should have worked. [Here](https://github.com/darkreader/darkreader/blob/master/src/inject/dynamic-theme/watch.ts#L165) it is triggered on attribute change and when the condition fulfills (style becomes manageable when media is no longer "print"), it should process it. Try finding out what's the problem. Maybe changes of media through JS are not reflected in attribute value.

Reply:
> Ehmm, looks like a hack. Is it a common problem for these websites? Why do they do this? Like the styles are not shown until they finish loading? I thought the browser optimizes it. Maybe it fixes the problem for some IE.

https://github.com/darkreader/darkreader/issues/2729#issuecomment-640201661
https://css-tricks.com/the-simplest-way-to-load-css-asynchronously/
It's probably to improve FCP(First contenfull Paint) aka Eliminate Render-Blocking Resources to speed it up, it's one of the ways to do this.

> Let's find out if there's a less hacky way. E.g. watching for media attribute change should have worked. Here it is triggered on attribute change and when the condition fulfills (style becomes manageable when media is no longer "print"), it should process it. Try finding out what's the problem. Maybe changes of media through JS are not reflected in attribute value.

I'm not sure if we have a mutation observer on non-manage-able `elements` e.g. links, that's another workaround like you suggested.